snap/snapshots.go

```diff
@@ -27,27 +27,104 @@ import (
 	"path/filepath"
 	"strings"
 
-	"gopkg.in/yaml.v2"
-
 	"github.com/snapcore/snapd/strutil"
+	"gopkg.in/yaml.v2"
 )
 
 var osOpen = os.Open
 
-// SnapshotOptions contains the information provided by a snap package about
-// what files should be included in the snapshots.
+// SnapshotOptions describes the options available for snapshots.
+// The initial source of these options is a file in the snap package.
+// In addition, options can be modified with dynamic requests via REST API.
 type SnapshotOptions struct {
 	// ExcludePaths is the list of file and directory patterns that need to be
 	// excluded from a snapshot. At the moment the only supported globbing
 	// character is "*", which stands for any sequence of characters other than
 	// "/".
-	ExcludePaths []string `yaml:"exclude"`
+	ExcludePaths []string `yaml:"exclude" json:"exclude,omitempty"`
```

Comment:
For the json tag we try to stay close to the var name, so this is probably `exclude-paths`, i.e. ` json:"exclude,-pathsomitempty"``

Reply:
Yes, I understand with what you are saying, however @pedronis commented in the review in preference on using "exclude" because it is already established for the snapshots.yaml. This will require spec change.

Reply:
Thanks! I guess in this case a short comment is good why var name and json differ but maybe @pedronis also has an opinion here :)

Reply:
This is essentially preexisting, if we want to change ExlcudePaths => Exclude (we cannot change the tag part) depending how churny it is we might have to do it in a separate PR.  I don't have a very strong opinion we try to make the golang field name and tag field name match, but we have a few exceptions here and there.

Reply:
Thanks. It's 15 hits in 2 code files and 1 unit test file. I will rename the field.

Reply:
`ExcludePaths` renamed to `Exclude` to match the tag as agreed in the spec meeting 27 Feb.